static/src/javascripts/projects/commercial/modules/dfp/Advert.js

```diff
@@ -16,7 +16,7 @@ type Resolver = (x: boolean) => void;
  * One size       - `data-mobile="300,50"`
  * Multiple sizes - `data-mobile="300,50|320,50"`
  */
-const createSizeMapping = (attr: string): Array<AdSize> =>
+const createSizeMapping = (attr: string): Array<SingleSize> =>
```

Comment:
Yes to the literal types, definitely. Seems like a great match.

Not so sure about the look of `GuAdSize`, i'm not sure if we really intend to specify that a parameter of the intersection type above is a useful contract to be used by other modules. Modelling-wise, I think it makes most sense  if `ad-sizes:AdSize` either contains a `SingleSize | MultiSize` field, or the more general types in `types.js` are modified to be usable in all modules (that can be achieved by merging the fields `width`, `height`, `switchUnitId` and `toString` into the general type. Might be a bit big for this PR?